### PR TITLE
Updated OutboundType to include comment for Swagger generation - Updated Swagger Readmes

### PR DESCRIPTION
**Pull request by carlowisse**
### Which issue this PR addresses:
[JIRA LINK](https://issues.redhat.com/browse/ARO-3118)

### What this PR does / why we need it:
These changes fix the linting issues required to pass upastream swagger specs in azure-rest-api-specs: [LINK TO UPSTREAM PR](https://github.com/Azure/azure-rest-api-specs/pull/23772)

### Test plan for issue:
- Ran `make client`
- Ran `make generate`
- E2E

### Is there any documentation that needs to be updated for this PR?
No, no process changes.

### CODE REVIEW

**Review by bennerv — APPROVED** [review]

**Review by bennerv — COMMENTED** [review]
Actually, I'm wondering if we want Jerome's input on the description of these, or if they don't care.  

We could paraphrase what Openshift talks about for their definition as well:

> The outbound routing strategy used to connect your cluster to the internet. If you are using user-defined routing, you must have pre-existing networking available where the outbound routing has already been configured prior to installing a cluster. The installation program is not responsible for configuring user-defined routing.

**Review by s-amann — APPROVED** [review]

File: pkg/api/v20230401/openshiftcluster.go [hunk 1]
```diff
@@ -126,8 +126,10 @@ type ServicePrincipalProfile struct {
 	ClientSecret string `json:"clientSecret,omitempty" mutable:"true"`
 }
 
+// OutboundType represents the type of routing a cluster is using.
 type OutboundType string
 
+// OutboundType constants.
```
Comment (jhoreman) [inline pkg/api/v20230401/openshiftcluster.go:132]: Hey Carlo, thanks for your work on this. Does this need to be backported to any of the other pkg/api versioned directories?

**Review by tony-schndr — COMMENTED** [review]
Just realized I missed the description for this field, thanks for adding it Carlo!

**Comment by tony-schndr** [comment]
Should `pkg/api/v20230401/openshiftcluster_example.go` be updated to include outboundType as well?

**Review by bennerv — DISMISSED** [review]

**Review by jhoreman — APPROVED** [review]

File: python/client/azure/mgmt/redhatopenshift/v2023_04_01/models/_azure_red_hat_open_shift_client_enums.py
Comment (jboutaud) [inline python/client/azure/mgmt/redhatopenshift/v2023_04_01/models/_azure_red_hat_open_shift_client_enums.py:1]: "OutboundType represents the type of egress routing a cluster is using"

Comment (bennerv) [reply python/client/azure/mgmt/redhatopenshift/v2023_04_01/models/_azure_red_hat_open_shift_client_enums.py:1]: > The outbound routing strategy used to provide your cluster egress to the internet.

@carlowiosse talked to @jboutaud and the above is the wording we'd like.

**Comment by jboutaud** [comment]
lgtm

**Review by bennerv — APPROVED** [review]

**Pull request merged**